src/App.tsx

```diff
@@ -1,56 +1,102 @@
-import React from 'react';
+import React, {
+  useCallback,
+  useMemo,
+  useState,
+  useEffect,
+  useRef,
+} from 'react';
 import './App.scss';
+import debounce from 'lodash.debounce';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/peopleList';
+
+const filterPeople = (query: string) => {
+  const normalizedQuery = query.trim().toLowerCase();
+
+  return peopleFromServer.filter(person => {
+    const normalizedName = person.name.trim().toLowerCase();
+
+    return normalizedName.includes(normalizedQuery);
+  });
+};
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [delay] = useState(1000);
```

Comment:
The delay is hardcoded and is not customizable as required in the task. Consider passing delay as a prop to the component or getting it from component's state.